torchqdynamics/odeint.py

```diff
@@ -11,6 +12,29 @@
 from .adaptive import DormandPrince45
 from .solver_options import AdaptiveStep, Dopri45, FixedStep
 from .solver_utils import add_tuples, bexpect, none_to_zeros_like
+from .types import OperatorLike
+
+
+def H_dependent(func):
+    """
+    Handles caching for functions that only depend on the Hamiltonian.
+    The functions must take as single argument the time and be decorated
+    by `@H_cached`. For instance::
+        @H_cached
+        def H_squared(self, t):
+            return self.H(t) @ self.H(t)
+    """
```

Comment:
See https://github.com/PierreGuilmin/torchqdynamics/pull/37#discussion_r1148350366 for the doc style. I need to edit the `CONTRIBUTING.md` with all the new conventions.

Reply:
Not sure what you want to change here